### PR TITLE
docs-infra: search tidy up

### PR DESCRIPTION
**Pull request by petebacondarwin**
This is a follow on from https://github.com/angular/angular/pull/41368

### CODE REVIEW

File: aio/tools/transforms/angular-base-package/processors/generateKeywords.js [hunk 1]
```diff
@@ -120,27 +120,28 @@ function isString(value) {
...
+  const rawTokens = text.split(/[\s/]+|<\/?[a-z]+(?:\s+\w+(?:="[^"]+")?)*>/ig);
   const tokens = [];
-  for(let token of rawTokens) {
+  for (let token of rawTokens) {
     token = token.trim();
 
-    // Strip off unwanted trivial characters
-    token = token.replace(/^[_\-"'`({[<$*)}\]>.]+/, '').replace(/[_\-"'`({[<$*)}\]>.]+$/, '');
+    // Trim unwanted trivial characters from the start and end of the token
+    token = token.replace(/^[\s_"'`({[<$*)}\]>.,-]*([\w.-]*[^\W_])[\s_"'`({[<$*)}\]>.,-]*$/, '$1');
```
Comment (gkalpak) [inline aio/tools/transforms/angular-base-package/processors/generateKeywords.js:129]: FWIW, I find this regex a little difficult to read. Especially the `[^\W_]` part made my head spin 🤯

A couple of (optional) suggestions:
- Switch back to the two `replace()` calls (one for `/^.../` and one for `/...$/`).
- Replace `[^\W_]` with `[a-zA-Z0-9]`.
- Change the regex to: 
    ```suggestion
        token = token.replace(/^[\s_"'`({[<$*)}\]>.,-]*([\w.-]*?)[\s_"'`({[<$*)}\]>.,-]*$/, '$1');
    ```

**Review by gkalpak — APPROVED** [review]
Nice! Just a few nits - otherwise lgtm 🚀 

Also there's a commit message type[sic] (😛): `fix type` --> `fix typo`

Reviewed-for: global-docs-approvers

File: aio/tools/transforms/angular-base-package/processors/generateKeywords.js [hunk 2]
```diff
@@ -38,7 +38,7 @@ module.exports = function generateKeywordsProcessor(log) {
 
       const filteredDocs = docs
           // We are not interested in some docTypes
-          .filter(doc => !docTypesToIgnore.has(doc.docType))
+          .filter(doc => doc.docType !== undefined && !docTypesToIgnore.has(doc.docType))
```
[the remark this answers is not in the document]
Comment (petebacondarwin) [reply aio/tools/transforms/angular-base-package/processors/generateKeywords.js:41]: I think it is six of one and half-a-dozen of another.
Happy to change this if you think it is better that way.

[the remark this answers is not in the document]
Comment (gkalpak) [reply aio/tools/transforms/angular-base-package/processors/generateKeywords.js:41]: Not sure myself. I'm fine either way, so let's leave it as is :grin:

**Pull request closed without merging**